[storage] Benchmarking DSLab disk vs SimGrid disk (#88)
osukhoroslov/dslab: examples-other/simgrid/storage/README.md

# Storage example & benchmark

> This is a SimGrid example corresponding to [storage-disk-benchmark](../../../examples/storage-disk-benchmark). It starts given amount of disk read requests, randomly distributed on multiple disks. Request size and start time are randomized too.

Reviewer: ```suggestion
This is a SimGrid example corresponding to [storage-disk-benchmark](../../../examples/storage-disk-benchmark). It executes the specified number of disk read requests with random size and start time. The requests are randomly distributed across the specified number of disks.
```

## Build

Run `make storage` from build directory.

## Arguments
 
| name               | meaning                     | default value               |
|--------------------|-----------------------------|-----------------------------|
| `--requests`       | Number of requests (>= 1)   | `1`                         |
| `--disks`          | Number of disks (>= 1)      | `1`                         |
| `--max-size`       | Maximal size (>= 1)         | `1000000006`                |

> | `--max-start-time` | Maximal activity start time | `0`, so all will start at `0` |

Reviewer: ```suggestion
| `--max-start-time` | Maximal request start time | `0`, so all will start at `0` |
```

## Run

Example:

```
./bin/storage \
    --log=root.thres:info \
    --requests 10 \
    --disks 10 \
    --max-size 100 \
    --max-start-time 100
```

> ## Comparing script

Reviewer: ```suggestion
## Comparing results with DSLab
```

> There is a [script](./compare-with-dslab.py) for comparing results evaluated by SimGrid and DSLab. Argument format is the same as for examples. You should set environment variable `DSLAB_BASE_DIR` before using and build both DSLab and SimGrid examples.

Reviewer: ```suggestion
There is a [script](./compare-with-dslab.py) for running both SimGrid and DSLab implementations and comparing their results. Argument format is the same as for examples. You should set environment variable `DSLAB_BASE_DIR` before using and build both DSLab and SimGrid examples.
```
